woof-code/rootfs-skeleton/etc/rc.d/rc.sysinit

```diff
@@ -149,6 +149,17 @@ echo -n "Making the filesystem usable..." >/dev/console
 # busybox mount /proc before remounting / as rw (full install)
 busybox mount -t proc none /proc 2>/dev/null; STATUS=$((STATUS+$?))
 
+mkdir /sys 2>/dev/null
+busybox mount -t sysfs none /sys ;STATUS=$((STATUS+$?))
+
+#Update microcode if possible
+if [ "$(uname -m | grep -E '^x86_64|^amd64|^i.86')" != "" ] && [ "$(which lscpu)" != "" ]; then
```

Comment:
Can this be done with /proc/cpuinfo instead of lscpu? I'm doing some work to speed up Puppy's boot on x86_64 and this is exactly the kind of things I'm looking for

Reply:
Any ideas on how to detect physical or virtual CPU by parsing /proc/cpuinfo?

Reply:
Can't you just let it fail if it's a VM?

Reply:
Nope not a good idea. I read on Arch Wiki that microcode loading was for physical cpu only.

Reply:
`dmidecode`?

Reply:
`grep -qm1 'hypervisor' /proc/cpuinfo && do_it || dont_do_it`

Reply:
@01micko I made some revision on the script